lib/wordnet/synset.rb

```diff
@@ -114,13 +217,48 @@ def expanded_hypernym
       while parent
         break if list.include? parent.pos_offset
         list.push parent.pos_offset
-        parent = parent.parent
+        parent = parent.hypernym
       end
 
       list.flatten!
       list.map! { |offset| Synset.new(@pos, offset)}
     end
 
+    # Get the entire hypernym tree (from this synset all the way up to +entity+) as an array.
+    def expanded_hypernyms
+      parents = hypernyms
+      list = []
+      return list unless parents
+
+      while parents.length > 0
+        parent = parents.pop
+        next if list.include? parent.pos_offset
+        list.push parent.pos_offset
+        parents.push *parent.hypernyms
+      end
+
+      list.flatten!
+      list.map! { |offset| Synset.new(@pos, offset)}
+    end
+
+    def expanded_hypernyms_depth
+      parents = hypernyms.map{|hypernym| [hypernym, 1]}
+      list = []
+      out = []
+      return list unless parents
```

Comment:
this will never happen since empty array is truthy ... should this be if `parents.empty?` ... then we are missing a test ...